src/insert.rs

```diff
@@ -179,7 +180,15 @@ impl SupabaseClient {
         };
 
         if response.status().is_success() {
-            Ok(())
+            let res_text: String = match response.text().await {
+                Ok(text) => text,
+                Err(e) => return Err(format!("Failed to get response text: {}", e)),
+            };
+            let id: String = match serde_json::from_str::<Value>(&res_text) {
+                Ok(json) => json["id"].to_string(),
```

Comment:
This makes an assumption that the table has an `id` column.
As Reacher says: `Assumptions kill!`
A table can use any name for its primary key! The key might even be a complex one, i.e. a combination of several columns.
IMO this method should return the whole record.